INSTALL.md

```diff
@@ -1,95 +1,508 @@
-# Prerequisites
+# Instructions
 
-| Linux | macOS | Windows 10
-| - | - | -
-| none | [Xcode Command Line Tools Package][xcode] | [Windows Terminal][terminal] and [Windows Subsystem for Linux (WSL)][wsl]
+These instructions explain how to set up the tools required to build **pokeemerald**, which assembles the source files into a ROM.
 
-[xcode]: https://developer.apple.com/library/archive/technotes/tn2339/_index.html
-[terminal]: https://docs.microsoft.com/windows/terminal/get-started
-[wsl]: https://docs.microsoft.com/windows/wsl/install-win10
+If you run into trouble, ask for help on Discord or IRC (see [README.md](README.md)).
 
-Independently from the specific OS, make sure that the `gcc`, `g++`, `make`, `git`, and `libpng-dev` packages or their equivalents are installed and accessible to the development tools that are used by the project (this means that, for example, on Windows, the packages have to be installed in the WSL environment). The package names and installation methods may vary with each OS.
+## Windows
+Windows has instructions for building with three possible terminals, if users encounter unexpected errors in following instructions for one of the terminals. These instructions are:
+- [Windows 10 (WSL1)](#windows-10-wsl1) (**Fastest, highly recommended**, Windows 10 only)
+- [Windows (msys2)](#windows-msys2) (Second fastest)
+- [Windows (Cygwin)](#windows-cygwin) (Slowest)
 
-Install the devkitARM toolchain of devkitPro as per [the instructions on their wiki](https://devkitpro.org/wiki/devkitPro_pacman). On Windows, follow the Linux instructions inside WSL as any steps about the Windows installer do not apply.
+Unscientific benchmarks suggest **msys2 is 2x slower** than WSL1, and **Cygwin is 5-6x slower** than WSL1.
 
-**Debian-based distro users:** This applies to Debian, Ubuntu, and similar distros, including in WSL. If necessary, install the `libarchive13`, `pkg-config`, and `gdebi-core` packages to be able to install devkitPro.
+> Note for advanced users: **WSL2** is an option and is even faster than **WSL1** if files are stored on the WSL2 file system, but some tools may have trouble interacting with the WSL2 file system over the network drive. For example, tools which use Qt versions before 5.15.2 such as [porymap](https://github.com/huderlem/porymap) may [have problems with parsing the `\\wsl$` network drive path](https://bugreports.qt.io/browse/QTBUG-86277).
```

Comment:
This will be outdated for porymap soon as it moves to Qt6. Are there are any other notable tools for which this is true?

Reply:
I tested using a bunch of tools in the WSL1 network drive, and apparently only porymap has that Qt issue.
Graphics tools tested:
- Microsoft Paint
- Paint.net
- GIMP
- GraphicsGale
- Usenti

Emulators:
- no$gba (source file inlining isn't normally detected, but if you copy the file to the same directory as the elf it'll work. I'm not sure if this is an issue with the wsl drive or just the source inlining detection failing in general. **This should probably be tested**)
- Bizhawk

Misc:
- Tilemap Studio

Note that while I didn't do any real tests, I tried using GitHub Desktop with a repository in the WSL1 Drive, and it seemed slower (although I didn't even do a comparison with the Windows Drive). This should be looked at as well.